cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py

```diff
@@ -1700,8 +1701,27 @@ def _write_openstack_snap_config(component):
             openstack.endpoint_tls_ca
         ).decode('utf-8'))
         lines.append('ca-file = {}'.format(str(cloud_endpoint_ca_path)))
-
-    cloud_config_path.write_text('\n'.join(lines))
+    if any([openstack.subnet_id,
+            openstack.floating_network_id,
+            openstack.lb_method,
+            openstack.manage_security_groups]):
+        lines.append('')
+        lines.append('[LoadBalancer]')
+    if openstack.subnet_id:
+        lines.append('subnet-id = {}'.format(openstack.subnet_id))
+    if openstack.floating_network_id:
+        lines.append('floating-network-id = {}'.format(
+            openstack.floating_network_id))
+    if openstack.lb_method:
+        lines.append('lb-method = {}'.format(
+            openstack.lb_method))
+    if openstack.manage_security_groups:
+        lines.append('manage-security-groups = {}'.format(
+            openstack.manage_security_groups))
+        lines.append('node-security-group = {}'.format(
+            openstack.node_security_group))
+
+    cloud_config_path.write_text(''.join('{}\n'.format(l) for l in lines))
```

Comment:
Curious why you use this instead of '\n'.join(lines)

Reply:
To get a trailing newline.